app/app/BaseUser.php

```diff
@@ -268,6 +268,14 @@ public static function confirmEmail(string $email,string $confirmCode)
 
 		return false;
 	}
+
+	// check if user's email verificated while loging in
+	public static function checkEmail(string $email) {
+		$matching_user = User::where('email', '=', $email)->first();
+		if(!is_null($matching_user->email_verification_time)) return true;
```

Comment:
Instead of using an if-statement like this, you could shorten this to 
```
return !is_null($matching_user->email_verification_time);
```

This shortened version seems a little easier to read.